File: 240315/블럭쌓는 명령2/block-stacking-commands2.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System. in);

        int n = sc.nextInt();
        int k = sc.nextInt();
        int[] arr = new int[n];

        for(int i = 0; i < k; i++){
            int A = sc.nextInt();
            int B = sc.nextInt();

            for(int j = A-1; j < B; j++){
                arr[j]++;

            }

        }
        int max = 0;

        for(int i = 0; i < n; i++){
            if(max < arr[i]){
                max = arr[i];

            }    

        }

        System.out.printf("%d", max);


        

        // 여기에 코드를 작성해주세요.
    }
}

File: 240727/블럭쌓는 명령2/block-stacking-commands2.py
n,k = map(int,input().split())
arr = [0 for _ in range(n+1)]
for _ in range(k):
    s,e = map(int,input().split())
    for i in range(s,e+1):
        arr[i]+=1
print(max(arr[1:]))